EIPS/eip-6372.md

```diff
@@ -40,20 +51,18 @@ This function returns the current timepoint. It could be `block.timestamp`, `blo
       type: uint48
 ```
 
-### CLOCK_MODE
+#### CLOCK_MODE
 
-This function returns a string describing the clock the contract is operating on.
+This function returns a machine-readable string description of the clock the contract is operating on.
+
+This string MUST be formatted like a URL query string (a.k.a. `application/x-www-form-urlencoded`), decodable in standard JavaScript with `new URLSearchParams(CLOCK_MODE)`.
 
 - If operating using **block number**:
-  - If the block numbers are those of the `NUMBER` opcode (`0x43`), then this function SHOULD return `mode=blocknumber&from=default`.
-  - If it is any other block number, then this function SHOULD return `mode=blocknumber&from=<CAIP2ID>`, where `<CAIP2ID>` is a CAIP-2 Blockchain ID such as `eip155:1`.
-- If operating using **timestamp**, then this function SHOULD return `mode=timestamp`.
+  - If the block number is that of the `NUMBER` opcode (`0x43`), then this function MUST return `mode=blocknumber&from=default`.
+  - If it is any other block number, then this function MUST return `mode=blocknumber&from=<CAIP-2-ID>`, where `<CAIP-2-ID>` is a CAIP-2 Blockchain ID such as `eip155:1`.
+- If operating using **timestamp**, then this function MUST return `mode=timestamp`.
 - If operating using any other mode, then this function SHOULD return a unique identifier for the encoded `mode` field.
```

Comment:
I leave this one as "SHOULD" because it's not feasible that someone will check all existing contracts to make sure their custom mode is unique.